src/lib/mina_base/parties.ml

```diff
@@ -294,6 +294,9 @@ let nonce (t : t) : Account.Nonce.t = (fee_payer_party t).data.predicate
 
 let fee_token (_t : t) = Token_id.default
 
+let fee_excess (t : t) =
+  Fee_excess.of_single (fee_token t, Currency.Fee.Signed.of_unsigned (fee t))
```

Comment:
This seems dangerous: the actual fee excess corresponding to a parties transaction is either the fee -- in the case of failure -- or the fee + whatever positive excess is left. Recommend removing this.

Reply:
I thought adding a fee payer explicitly was to make sure we have a fixed fee? and that we would be checking if the final fee excess will equal to the fee. This is the case with payments.
So does it mean `fee_excess` is still `fee_excess_lower_bound`?  we need this in transaction pool to check if the fee is >= min required. 
Also, in the case of non--default tokens the excess (apart from fee payer delta) should be zero, right?
What's the case where we'd see parties (in `other_parties`) that want to pay fees in addition to the fee payer which is required to get included in to the pool? Block producers would also be prioritizing txns based on this lower bound as they won't know the exact fee excess until it is applied. So not sure what's the use case for this

Reply:
`fee_excess >= fee_excess_lower_bound`. When the excess is non-zero after applying the remaining parties, we have to do something, and currently it goes to the block producer. If this should change, lets discuss in a different issue or PR, and for now consider the excess 'unknowable' when the applicability of a transaction is unknown, as would have been correct since the fee-payer change.